framework/wazuh/core/batcher/batcher.py

```diff
@@ -0,0 +1,196 @@
+        - Handling shutdown signals.
+        """
+        # Register signal handlers
+        self._shutdown_event = asyncio.Event()
+        loop = asyncio.get_running_loop()
+        loop.add_signal_handler(signal.SIGINT, self._handle_signal, signal.SIGINT)
+        loop.add_signal_handler(signal.SIGTERM, self._handle_signal, signal.SIGTERM)
+
+        try:
+            while not self._shutdown_event.is_set():
+                done, pending = await asyncio.wait(
+                    [self._get_from_queue(), self._timer.wait_timeout_event()],
+                    return_when=asyncio.FIRST_COMPLETED
+                )
+
+                # Process completed tasks
+                for task in done:
+                    if isinstance(task.result(), Message):
+                        message = task.result()
+
+                        if self._buffer.get_length() == 0:
+                            self._timer.create_timer_task()
+
+                        self._buffer.add_message(message)
+
+                        if self._buffer.check_count_limit() or self._buffer.check_size_limit():
+                            self.create_flush_buffer_task()
+                            self._timer.reset_timer()
+                    else:
+                        # Cancel the reading task if it is still pending
+                        for p_task in pending:
+                            p_task.cancel()
+
+                        self.create_flush_buffer_task()
+                        self._timer.reset_timer()
+
+        except Exception as e:
+            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+            logger.error(f"Error in batcher loop:\n{''.join(tb_str)}")
+        finally:
+            # Ensure all tasks are properly cancelled
+            for task in asyncio.all_tasks():
+                if task is not asyncio.current_task():
+                    task.cancel()
+            await asyncio.gather(*asyncio.all_tasks(), return_exceptions=True)
+
+    def _handle_signal(self, signal_number):
+        """
+        Handles shutdown signals by setting the shutdown event.
+
+        Parameters
+        ----------
+        signal_number : int
+            The signal number indicating the type of signal received (e.g., SIGINT, SIGTERM).
+        """
+        logger.info(f"Batcher - Received signal {signal_number}, initiating shutdown.")
+        self._shutdown_event.set()
+
+
+class BatcherProcess(Process):
+    """
+    A process that runs a Batcher instance. This class is used to execute the Batcher in a separate process.
+
+    Parameters
+    ----------
+    q : MuxDemuxQueue
+        The MuxDemuxQueue instance for message multiplexing and demultiplexing.
+    config : BatcherConfig
+        Configuration parameters for batching, such as maximum elements and size.
+    """
+    def __init__(self, q: MuxDemuxQueue, config: BatcherConfig):
+        super().__init__()
+        self.q = q
```

Comment:
I would prefer `queue`, to be more descriptive and differentiate it from "query"

```suggestion
    queue : MuxDemuxQueue
        The MuxDemuxQueue instance for message multiplexing and demultiplexing.
    config : BatcherConfig
        Configuration parameters for batching, such as maximum elements and size.
    """
    def __init__(self, queue: MuxDemuxQueue, config: BatcherConfig):
        super().__init__()
        self.queue = queue
```

Reply:
Fixed in https://github.com/wazuh/wazuh/commit/47e4508357e0fb53f499847d6a791a0f4c86f15b.